pandas/io/formats/csvs.py

```diff
@@ -157,13 +171,19 @@ def save(self):
             self._save()
 
         finally:
-            # GH 17778 handles zip compression for byte strings separately.
-            buf = f.getvalue()
-            if path_or_buf:
-                f, handles = _get_handle(path_or_buf, self.mode,
-                                         encoding=encoding,
-                                         compression=self.compression)
-                f.write(buf)
+            if is_zip:
+                # GH 17778 handles zip compression separately.
+                buf = f.getvalue()
+                try:
+                    self.path_or_buf.write(buf)
```

Comment:
They say that it's sometimes easier to ask for forgiveness than it is for permission.

That being said, your treatment of `self.path_or_buf.write` is inconsistent here compared to:

https://github.com/pandas-dev/pandas/pull/21478/files#diff-eaa887c826bfb361d98db1ddb668c7deR150

In addition, the logic seems somewhat similar.  Can we potentially deduplicate this logic?

Reply:
see your point. changed to if else to make it consistent.